server/src/main/java/com/google/musicanalysis/site/YoutubeServlet.java

```diff
@@ -15,39 +15,106 @@
 import javax.servlet.http.HttpServletResponse;
 import javax.servlet.http.HttpSession;
 import javax.servlet.annotation.WebServlet;
-
+import java.util.HashMap;
 
 @WebServlet("/api/youtube")
 public class YoutubeServlet extends HttpServlet {
-    @Override
-    protected void doGet(HttpServletRequest req, HttpServletResponse res) 
+    /**
+     * makes http request of youtube api to retrieve topics of liked videos, 
+     *  gets json string of youtube response
+     * @param apiKey youtube api key
+     * @param accessToken youtube access token from login
+     * @return JSON string of youtube response of liked video topics
+     * @throws ServletException
+     * @throws IOException
+     */
+    protected String getYoutubeRes(String apiKey, String accessToken) 
         throws ServletException, IOException {
-        // retrieve Youtube API key and access token 
-        String API_KEY = Secrets.getSecretString("YOUTUBE_API_KEY");
-        HttpSession session = req.getSession();
-        var accessToken = session.getAttribute("oauth-access-token-youtube");
-        if (accessToken == null) {
-            res.setStatus(401);
-            return;
-        }
-    
         // make http request to youtube API
-        var youtubeParam = new URLEncodedBuilder()
+        URLEncodedBuilder youtubeParam = new URLEncodedBuilder()
```

Comment:
Ah I see how URLEncodedBuilder works now. This is a cool way to do it bc it takes care of the encoding for you.